lib/active_support/cache/database_store.rb

```diff
@@ -120,6 +131,25 @@ def delete_entry(key, _options = nil)
         @model.where(key: key).destroy_all
       end
 
+      def write_multi_entries(hash, **_options)
+        entries = hash.map {|key, entry| { key: key, created_at: Time.zone.now, **entry_attributes(entry) } }
+
+        # In rails 7, we can use update_only not to do anything. But for the sakes of compatibility, we don't use any additional parameters.
+        @model.upsert_all(entries)
+      end
+
+      def entry_attributes(entry)
+        expires_at = Time.zone.at(entry.expires_at) if entry.expires_at
+        attributes = case entry.value
+                     when Numeric # Integer and float entries do not warrant compression
```

Comment:
I would go further, anything under 1024 bytes doesn't warrant compression in my opinion. short strings will not benefit from it. I suggest you do:

```rb
attributes = compressed_attributes(Marshal.dump(entry.value))
```